cambiatus/frontend
# Enhancement/add tests

## What issue does this PR close
Closes #535 

## Changes Proposed ( a list of new changes introduced by this PR)
- Layout initial testing structure
- Add `elm-test` to our `pre-push` hook
- Add `elm-test` to our CI/CD pipeline on github
- Create examples of how to test with `elm-test`
- Create some helper functions for testing
- Define a baseline for what tests should look like and how they should be organized in our project
- Introduce some tests for already-existing modules/functions

Note: This PR might not cover every single thing that needs testing right now. It's meant to cover some things, and establish examples for future us to use as a reference. This PR adds many tests, but only because we didn't have any tests, so this should be an exception! In order to maintain tests up to date, pull requests after this one should introduce tests alongside their enhancements/features/bug-fixes

## How to test ( a list of instructions on how to test this PR)

Well, we can't write tests for tests :sweat_smile:, so look over the overall structure and organization, check if you agree with it or if we could improve on something. Check if the tests make sense, cover the functions appropriately, and actually pass. Tests should be run on github alongside our other checks, like `elm-review` and build. The actual code of the app shouldn't change (too much), which means we're not really risking breaking the app with this PR
